pkg/oob/oob.go

```diff
@@ -0,0 +1,196 @@
+package oob
+
+import (
+	"context"
+
+	"github.com/bmc-toolbox/bmclib/devices"
+	"github.com/bmc-toolbox/bmclib/discover"
+	v1 "github.com/tinkerbell/pbnj/pkg/api/v1"
+	"github.com/tinkerbell/pbnj/pkg/logging"
+)
+
+// User management methods
+type User interface {
+	Create(ctx context.Context, in v1.CreateUserRequest) (result string, err *Error)
+	Update(ctx context.Context, in v1.UpdateUserRequest) (result string, err *Error)
+	Delete(ctx context.Context, in v1.DeleteUserRequest) (result string, err *Error)
+}
+
+// Machine management methods
+type Machine interface {
+	BootDevice(ctx context.Context, in *v1.DeviceRequest) (result string, err *Error)
+	Power(ctx context.Context, status chan string, in *v1.PowerRequest) (result string, err *Error)
+}
+
+// BMC management methods
+type BMC interface {
+	Reset(ctx context.Context, in v1.ResetRequest) (result string, err *Error)
+	NetworkSource(ctx context.Context, in v1.NetworkSourceRequest) (result string, err *Error)
+}
+
+// Conn details for connecting
+type Conn struct {
+	Log logging.Logger
+}
+
+// Error for all bmc actions
+type Error struct {
+	*v1.Error
+}
+
+// BootDevice functionality for machines
+func (c *Conn) BootDevice(ctx context.Context, in *v1.DeviceRequest) (result string, err *Error) {
+	l := c.Log.GetContextLogger(ctx)
+	l.V(0).Info("not implemented")
+	return result, err
+}
+
+// Power functionality for machines
+func (c *Conn) Power(ctx context.Context, status chan string, in *v1.PowerRequest) (string, *Error) {
+	l := c.Log.GetContextLogger(ctx)
+	l.V(0).Info("power state")
+	// TODO handle nil values
+	var result string
+	errMsg := &Error{
+		Error: &v1.Error{
+			Code:    0,
+			Message: "",
+			Details: nil,
+		},
+	}
+
+	if in.Authn == nil || in.Authn.Authn == nil {
+		msg := "no auth found"
+		errMsg.Code = 16
```

Comment:
16? There's no symbol for this magic number?

Reply:
ah sure, I'll update with a more user-friendly way to specify

Reply:
please let me know if the new update for this satisfies.